src/System.IO.FileSystem/tests/Directory/GetSetTimes.cs

```diff
@@ -1,118 +1,131 @@
-                    else
-                    {
-                        switch (timeProperty)
-                        {
-                            case TimeProperty.CreationTime:
-                                Directory.SetCreationTime(dirName, dt);
-                                break;
-                            case TimeProperty.LastAccessTime:
-                                Directory.SetLastAccessTime(dirName, dt);
-                                break;
-                            case TimeProperty.LastWriteTime:
-                                Directory.SetLastWriteTime(dirName, dt);
-                                break;
-                            default:
-                                break;
-                        }
-                    }
+                DateTime dt = new DateTime(2014, 12, 1, 12, 0, 0, tuple.Item3);
+                tuple.Item1(testDir.FullName, dt);
+                Assert.Equal(dt, tuple.Item2(testDir.FullName));
+            });
+        }
 
-                    DateTime actual, actualUtc;
-                    switch (timeProperty)
-                    {
-                        case TimeProperty.CreationTime:
-                            actual = Directory.GetCreationTime(dirName);
-                            actualUtc = Directory.GetCreationTimeUtc(dirName);
-                            break;
-                        case TimeProperty.LastAccessTime:
-                            actual = Directory.GetLastAccessTime(dirName);
-                            actualUtc = Directory.GetLastAccessTimeUtc(dirName);
-                            break;
-                        case TimeProperty.LastWriteTime:
-                            actual = Directory.GetLastWriteTime(dirName);
-                            actualUtc = Directory.GetLastWriteTimeUtc(dirName);
-                            break;
-                        default:
-                            throw new ArgumentException("Invalid time property type");
-                    }
+        [OuterLoop]
+        [Fact]
+        public void CreationSetsAllTimes()
+        {
+            string path = GetTestFilePath();
+            long beforeTime = DateTime.Now.Ticks;
+            long utcBeforeTime = DateTime.UtcNow.Ticks;
+            Task.Delay(2000).Wait(); //buffer to resolve creation time
 
-                    DateTime expected = dt.ToLocalTime();
-                    DateTime expectedUtc = dt.ToUniversalTime();
+            DirectoryInfo testDirectory = new DirectoryInfo(GetTestFilePath());
+            testDirectory.Create();
 
-                    if (dt.Kind == DateTimeKind.Unspecified)
-                    {
-                        if (setUtc)
-                        {
-                            expectedUtc = dt;
-                        }
-                        else
-                        {
-                            expected = dt;
-                        }
-                    }
+            Task.Delay(2000).Wait();
```

Comment:
Ahh. Yeah, you're right. The Task.Delay was only useful in the tests I removed that checked if writing to a file updated the LastWriteTime, since the underlying system calls took a while to actually update and a delay was necessary for that to complete. In the creation case that isn't necessary since there's no previous time to replace; either it has the right time or no time, not an old time.

Okay. Out of curiosity, why? Shouldn't DateTime.Now.AddSeconds(-1) just be an offset version of DateTime.Now.AddSeconds(-1)?